gtsfm/scene_optimizer.py

```diff
@@ -216,6 +219,10 @@ def _filter_dict_keys(dict: Dict[Any, Any], ref_dict: Dict[Any,Any]) -> Dict[Any
         # Save metrics to JSON and generate HTML report.
         auxiliary_graph_list.extend(save_metrics_reports(metrics_graph_list))
 
+        # Modify BA input and BA output to have point clouds and frustums aligned with x,y,z axes.
+        ba_input_graph = dask.delayed(align_gtsfm_data)(ba_input_graph)
```

Comment:
let's estimate the transform just once, since we need to make sure these two pose graphs are perfectly aligned with each other